lib/vnet/diag/routeconflict.go

```diff
@@ -0,0 +1,302 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package diag
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"net"
+	"net/netip"
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+var log = logutils.NewPackageLogger(teleport.ComponentKey, "vnet:diag")
+
+type RouteConflictConfig struct {
+	VnetIfaceName string
+	Routing       Routing
+	Interfaces    Interfaces
+}
+
+// Routing abstracts away platform-specific logic of obtaining routes with their destinations,
+// allowing running tests for the general logic behind RouteConflictDiag on any platform.
+type Routing interface {
+	GetRouteDestinations() ([]RouteDest, error)
+}
+
+// Interfaces abstracts away functions from the net package and calls to ifconfig, allowing mocking
+// interactions with them in tests.
+type Interfaces interface {
+	InterfaceByName(string) (*net.Interface, error)
+	InterfaceByIndex(int) (*net.Interface, error)
+	// InterfaceApp attempts to return the name of the app that created the interface given the name
+	// of the interface.
+	InterfaceApp(context.Context, string) (string, error)
+}
```

Comment:
Extra interfaces, pretty much only for tests. The list of files in `lib/vnet/diag` is kind of a mess as a result (.go, _darwin.go, _darwin_test.go, _other.go, _test.go). But what we get in return are tests for the basic logic that run within CI on Linux. `golang.org/x/net/route` is BSD-only.